junit-vintage-engine/src/test/java/org/junit/vintage/engine/discovery/VintageDiscoveryRequestResolverTests.java

```diff
@@ -58,19 +60,21 @@ void logsWarningWhenFilterExcludesClass() {
 			logRecord.getMessage());
 	}
 
-	public static class Foo {
```

Comment:
@sbrannen Otherwise they would not be discovered by the vintage engine. 
The isPotentialJunit4TestClass predicate contains a check: !candidate.isMemberClass();
So foo as a nested class would be filtered out making the tests fail.